<?xml version="1.0"?>

<launch>

  <arg name="enable_odom_tf" default="false"/>
  <arg name="odom_frame" default="odom"/>
  <arg name="odom_topic" default="odom"/>
  <arg name="cmd_topic" default="cmd_vel"/>
  <arg name="base_frame" default="base_link"/>
  <arg name="world_name" default=""/>

  <env name="GAZEBO_MODEL_PATH" value="$GAZEBO_MODEL_PATH:$(find racecar_gazebo)/models"/>
  <env name="GAZEBO_RESOURCE_PATH" value="$GAZEBO_RESOURCE_PATH:$(find racecar_gazebo)/models"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find racecar_gazebo)/worlds/world_$(arg world_name).world"/>
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
  </include>

  <include file="$(find racecar_gazebo)/launch/racecar_spawn.launch"/>

  <!-- ros_control racecar launch file -->
  <include file="$(find racecar_control)/launch/racecar_control.launch">
    <arg name="publish_tf" default="$(arg enable_odom_tf)"/>
    <arg name="odom_frame" default="$(arg odom_frame)"/>
    <arg name="odom_topic" default="$(arg odom_topic)"/>
    <arg name="base_frame" default="$(arg base_frame)"/>
  </include>

  <!-- Spawn the MUXs -->
  <arg name="racecar_version" default="racecar-v2" />
  <include file="$(find racecar)/launch/mux.launch" ns="vesc"/>

  <!-- Publish "better odom" topic that is normally generated by the particle filter -->
  <node name="better_odom" pkg="topic_tools" type="relay"
          args="/vesc/odom /pf/pose/odom" />
    
  <!--Launch the simulation joystick control-->
  <rosparam command="load" file="$(find racecar_gazebo)/config/keyboard_teleop.yaml" />
  <node pkg="racecar_gazebo" type="keyboard_teleop.py" name="keyboard_teleop" />

</launch>
